File: src/test/java/javabot/model/UserFactory.java
package javabot.model;

import javabot.Messages;
import org.pircbotx.PircBotX;
import org.pircbotx.User;
import org.pircbotx.UserChannelDao;

import javax.inject.Inject;
import javax.inject.Provider;

public class UserFactory {
    @Inject
    private Provider<PircBotX> ircBot;
    @Inject
    private Messages messages;

    public User createUser(final String nick, final String login, final String host) {
        return new TestUser(ircBot.get(), ircBot.get().getUserChannelDao(), messages, nick, login, host);
    }

}

File: src/test/kotlin/javabot/model/UserFactory.kt
package javabot.model

import javabot.Messages
import org.pircbotx.PircBotX
import org.pircbotx.User
import javax.inject.Inject
import javax.inject.Provider

public class UserFactory {
    @Inject
    protected lateinit var ircBot: Provider<PircBotX>
    @Inject
    protected lateinit var messages: Messages

    public fun createUser(nick: String, login: String, host: String): User {
        val bot = ircBot.get()
        return TestUser(bot, bot.userChannelDao, messages, nick, login, host)
    }

}
